go/adbc/driver/snowflake/binding.go

```diff
@@ -0,0 +1,141 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package snowflake
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow/go/v17/arrow"
+	"github.com/apache/arrow/go/v17/arrow/array"
+)
+
+func convertArrowToNamedValue(batch arrow.Record, index int) ([]driver.NamedValue, error) {
+	// see goTypeToSnowflake in gosnowflake
+	// technically, snowflake can bind an array of values at once, but
+	// only for INSERT, so we can't take advantage of that without
+	// analyzing the query ourselves
+	params := make([]driver.NamedValue, batch.NumCols())
+	for i, field := range batch.Schema().Fields() {
+		rawColumn := batch.Column(i)
+		params[i].Ordinal = i + 1
+		switch column := rawColumn.(type) {
+		case *array.Boolean:
+			params[i].Value = sql.NullBool{
+				Bool:  column.Value(index),
+				Valid: column.IsValid(index),
+			}
```

Comment:
Something for us to think about is that go1.22 added `sql.Null[T]` which might be useful to allow us to create a more generic implementation of this later on (since we officially still support go 1.21, we wouldn't be able to use it yet)